File: js/tombola.py
## 12,13,14,15,16,17,18,19,10
## 23,24,25,26,27,28,29,20
## 34,35,36,37,38,39,30
## 45,46,47,48,49,40
## 56,57,58,59,50
## 67,68,69,60
## 78,79,70
## 89,80
## 90

nEquipos = 10

equiposA=[]
equiposB=[]

i=0
n=nEquipos-1
while i<n:
    equiposA.append(i)
    equiposB.append(n)
    i+=1
    n-=1
print(equiposA)
print(equiposB)

n=0
i=0
j=0
equipoAux = []
while n<(nEquipos-1):

    print("Jornada ",n+1)
    while i<(nEquipos/2):
        print(equiposA[i],equiposB[j])
        j+=1
        i+=1
    i=(nEquipos//2)-1
    equipoAux = equiposB[0]
    del equiposB[0]
    aux = equiposA[i]
    equiposB.append(aux)
    del equiposA[i]
    equiposA.insert(1,equipoAux)
    j=0
    i=0
    n+=1

File: php/tombola.php
<?php

$consulta_datos = "SELECT * FROM equipos";

$conexion = conexion();

$datos = $conexion->query($consulta_datos);
$datos = $datos->fetchAll();
$equipos=array();
$foto='./img/equipo.png';

foreach($datos as $dato){
    if(is_file("./img/equipos/".$dato['foto'])){
        $foto = './img/equipos/'.$dato['foto'];
    }else{
        $foto='./img/equipo.png';
    }
    $equipos[]= array($dato['nombre'],$foto);
}
$conexion=null;

$nEquipos = count($equipos);

if ($nEquipos % 2 == 0) {

} else {

    $nEquipos += 1;
    $equipos[]= array("DESCANSO","./img/equipo.png");
}

$equiposA = [];
$equiposB = [];

$i = 0;
$n = $nEquipos - 1;
while ($i < $n) {
    $equiposA[] = $i;
    $equiposB[] = $n;
    $i += 1;
    $n -= 1;
}

$partidos="";

$A=0;
$B=0;
$n = 0;
$i = 0;
$j = 0;
$equipoAux = [];
while ($n < ($nEquipos - 1)) {
    $partidos.='<h1 class="titulo"> Jornada '  . ($n + 1) . '</h1>';

    while ($i < ($nEquipos / 2)) {

        $A=$equiposA[$i];
        $B=$equiposB[$j];

        $partidos.='<div class="item">
            <img src="'.$equipos[$A][1].'">
            <label>'.$equipos[$A][0].'</label> 
            <label>vs</label> 
            <label>'. $equipos[$B][0].'</label>
            <img src="'.$equipos[$B][1].'">
            </div>';

        $j += 1;
        $i += 1;
    }
    $i = floor($nEquipos / 2) - 1;
    $equipoAux = $equiposB[0];
    array_shift($equiposB);
    $aux = $equiposA[$i];
    $equiposB[] = $aux;
    array_splice($equiposA, $i, 1);
    array_splice($equiposA, 1, 0, $equipoAux);
    $j = 0;
    $i = 0;
    $n += 1;
}

echo $partidos;